File: src/frontend/lib/dto/caregiverDto.dart
import 'childDto.dart';

class CaregiverDto {
  final String caregiverId;
  final String name;
  final String email;
  final String? fcmToken;
  final List<ChildDto> children;

  CaregiverDto({
    required this.caregiverId,
    required this.name,
    required this.email,
    this.fcmToken,
    required this.children,
  });

  Map<String, dynamic> toJson() {
    return {
      'caregiverId': caregiverId,
      'name': name,
      'email': email,
      'fcmToken': fcmToken,
      'children': children.map((c) => c.toJson()).toList(),
    };
  }
}

File: src/backend/src/main/java/com/bouh/backend/model/Dto/caregiverDto.java
package com.bouh.backend.model.Dto;
import com.google.cloud.firestore.annotation.DocumentId;
import lombok.Data;
import java.util.List;

@Data //setters,getters and constructors
public class caregiverDto {
    @DocumentId
    private String caregiverId;

    private String name;
    private String email;
    private String fcmToken;
    private List<childDto> children;
}
